### PR TITLE
audit/manifests: log requests to Events

### PR DESCRIPTION
**Pull request by dgrisonnet**
Re-enable audit logs about Events to be able to identify misconfigured clients.

### CODE REVIEW

File: pkg/operator/apiserver/audit/manifests/base-policy.yaml [hunk 1]
```diff
@@ -6,8 +6,11 @@
     omitStages:
     - "RequestReceived"
     rules:
-    # Don't log requests for events
-    - level: None
+    # Log events at the metadata level. This is necessary to identify
+    # misconfigured clients. They could either spam mutating requests on Events
+    # or list all the Events in the cluster which are two operations that can
+    # lead to disruption.
+    - level: Metadata
```
Comment (p0lyn0mial) [inline pkg/operator/apiserver/audit/manifests/base-policy.yaml:13]: I think we could simply remove it. There is [a catch-all rule](https://github.com/openshift/library-go/blob/8052e81a61efbeabcd1ba0e660d9777a93fc6859/pkg/operator/apiserver/audit/manifests/default-rules.yaml#L10) which would be applied

File: pkg/operator/apiserver/audit/testdata/allrequestbodies.yaml [hunk 2]
```diff
@@ -8,8 +8,10 @@ omitManagedFields: true
 omitStages:
 - "RequestReceived"
 rules:
-# Don't log requests for events
-- level: None
+# Log events at the metadata level. This is necessary to identify misconfigured
+# clients. They could either spam mutating requests on Events or list all the
+# Events in the cluster which are two operations that can lead to disruption.
+- level: Metadata
```
Comment (p0lyn0mial) [inline pkg/operator/apiserver/audit/testdata/allrequestbodies.yaml:14]: I would also remove it. If a user requested all request bodies, let's give them what they want.

**Review by p0lyn0mial — COMMENTED** [review]
It would be useful to know the impact on storage, CPU, and memory in large clusters after enabling event logging. I understand that we are confident the impact will be negligible.

Could you also open a proof PR for kas and oas and manually validate if the policy was changed ?

**Review by dinhxuanvu — COMMENTED** [review]
/lgtm

**Comment by p0lyn0mial** [comment]
/approve

/hold 
in case you would like to test something.

**Comment by dgrisonnet** [comment]
Verified on the proof PR.

/unhold

**Pull request merged**